javaslang/src/main/java/javaslang/collection/Iterator.java

```diff
@@ -1392,24 +1392,37 @@ public T getNext() {
             final Iterator<T> that = this;
             return new AbstractIterator<T>() {
 
-                private T next = null;
+                private T next;
+                private boolean cached = false;
+                private boolean test = true;
```

Comment:
please rename `test` to reflect the intent better ...
`isFirstValue` or something?